// Copyright 2018 Delivery Hero Germany
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
//     Unless required by applicable law or agreed to in writing, software
//     distributed under the License is distributed on an "AS IS" BASIS,
//     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//     See the License for the specific language governing permissions and
//     limitations under the License.

syntax = "proto3";

package model;

import "google/protobuf/timestamp.proto";

message HeartbeatMessage {
    // Cluster details
    string cluster_name = 1;

    // Cluster metrics
    int32 AMResourceLimitMB = 2;
    int32 AMResourceLimitVCores = 3;
    int32 UsedAMResourceMB = 4;
    int32 UsedAMResourceVCores = 5;
    int32 AppsSubmitted = 6;
    int32 AppsRunning = 7;
    int32 AppsPending = 8;
    int32 AppsCompleted = 9;
    int32 AppsKilled = 10;
    int32 AppsFailed = 11;
    int32 AggregateContainersPreempted = 12;
    int32 ActiveApplications = 13;
    int32 AppAttemptFirstContainerAllocationDelayNumOps = 14;
    float AppAttemptFirstContainerAllocationDelayAvgTime = 15;
    int32 AllocatedMB = 16;
    int32 AllocatedVCores = 17;
    int32 AllocatedContainers = 18;
    int32 AggregateContainersAllocated = 19;
    int32 AggregateContainersReleased = 20;
    int32 AvailableMB = 21;
    int32 AvailableVCores = 22;
    int32 PendingMB = 23;
    int32 PendingVCores = 24;
    int32 PendingContainers = 25;

    google.protobuf.Timestamp Timestamp = 26;

    // Cluster specific information
    int32 NumberOfNodes = 27;

    // Cost field (in dollars)
    float Cost = 28;
}